src/main/java/frc/robot/subsystems/Navigation.java

```diff
@@ -0,0 +1,62 @@
+package frc.robot.subsystems;
+
+import java.io.IOException;
+
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.PhotonPoseEstimator;
+import org.photonvision.PhotonPoseEstimator.PoseStrategy;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.estimator.DifferentialDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.math.util.Units;
+import frc.robot.Constants;
+import frc.robot.subsystems.vision.Vision;
+
+public class Navigation {
+    private DifferentialDriveKinematics kinematics;
+    private DifferentialDrivePoseEstimator poseEstimator;
+    private PhotonPoseEstimator cameraPoseEstimator;
+    private Vision vision;
+    private AprilTagFieldLayout layout;
+
+    public Navigation () {  // REAL ROBOT
```

Comment:
It would be good to pass in the mode (SIM/REAL/REPLAY) for determining which vision to call

Reply:
would it be better to pass in or just read from constants like 2023 robot?

Reply:
I think reading the constants without passing them in is fine tbh